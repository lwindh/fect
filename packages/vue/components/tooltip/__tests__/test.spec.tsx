/**
 *  wrapper.emitted() does not register an emitted event
 *  if it was called by watch with immediate: true option
 *  see issue : https://github.com/vuejs/vue-test-utils-next/issues/259
 */

import { flushPromises, mount } from '@vue/test-utils'
import { Tooltip } from '..'
import { later } from '../../../tests'

const Wrapper = {
  components: {
    [Tooltip.name]: Tooltip
  },
  data() {
    return {
      visible: false
    }
  },
  template: `
  <div class="container">
   <fe-tooltip 
   ref="tooltipRef"
   v-model:visible="visible" content="testinfo" trigger="click">
    Test Data
   </fe-tooltip>
  </div>
  `
}

describe('Tooltip', () => {
  it('should be support slots content', () => {
    const wrapper = mount(Tooltip, {
      slots: {
        content: <div>test message</div>
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('bidirectional binding should be supported', async () => {
    const wrapper = mount(Wrapper, { attachTo: document.body })
    const {
      tooltipRef: { setTeleport }
    } = wrapper.vm.$refs as any
    await setTeleport('.container')

    const el = wrapper.find('.fect-tooltip')
    await el.trigger('click')
    await flushPromises()
    await later()
    expect(wrapper.vm.visible).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
