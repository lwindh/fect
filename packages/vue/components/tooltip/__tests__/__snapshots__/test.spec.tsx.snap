// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip bidirectional binding should be supported 1`] = `
<div class="container">
  <div class="fect-tooltip"> Test Data
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub>
    <div class="fect-tooltip__content  default" role="popup" style="--tooltip-icon-offset-x: 0px; --tooltip-icon-offset-y: 0px; top: -12px; left: 0px; transform: translate(-50%, -100%);">
      <div class="fect-tooltip__inner"><span class="fect-tooltip__arrow-icon default" style="left: 50%; bottom: 3px; transform: translate(-50%, 100%) rotate(-90deg);"></span>testinfo</div>
    </div>
  </transition-stub>
  <!---->
</div>
`;

exports[`Tooltip should be support slots content 1`] = `
<div class="fect-tooltip">
  <!---->
  <!--teleport start-->
  <!--teleport end-->
</div>
`;
